demo-transactions/src/main/kotlin/io/holixon/cqrshexagonaldemo/demoparent/transactions/command/application/usecase/DepositUsecase.kt

```diff
@@ -20,15 +21,23 @@ class DepositUsecase(
     override fun deposit(iban: Iban, amount: Amount) {
         logger().info { "deposit $amount to account ${iban.value}" }
 
+        require(amount.value >= BigDecimal.ZERO) {
+            logger().error {
+                "Amount must be positive"
+            }
+        }
+
         val account = accountOutPort.findAccount(iban)
```

Comment:
To be honest, I really don't like the naming, that we introduced here in the beginning. Reading "InPort" and "OutPort" in the usecase doesn't feel right, since it doesn't help to understand what is happening, but distracts by bringing "reading complexity".
I would like to have those things named by what they are meant to do, than by what they are. Like `AccountService` or `AccountRepo` instead of `AccountOutPort`. We can add `OutPort` and `InPort` annotations to the `framework` package, and mark the port-interfaces with those annotations, but that should be it.

Reply:
I understand what you mean and i will do the change. The outPort could be in fact called Services like *AccountService* instead of AccountOutPort. but i think the inPort could be kept no ? Since InPort immediately tells you what its purpose is.

Reply:
I would do it unison - name it after its domain/application purpose and add the architectural purpose as annotation. But this is not for now, we can do that later.